### ISSUE TITLE
Posture check for peer's Public IP-Address

### ISSUE THREAD

**Issue by piotrekzurek** [netbirdio/netbird#1706]
**Is your feature request related to a problem? Please describe.**
Like with GeoIP posture check it would be nice to have ability to "pin" a peer to be able to connect only from one certain public IP address to the network.
Currently I use IPSEC to do external unlocking of encrypted disk volumes on servers that works in a restricted area and yet the customer does not want to be in possession of a unlocking key (or just can't do it themselves). Having ability to pin peer's IP address to certain value would allow to not use IPSEC in this scenario. This is used to make the unlocking technically not possible (or at least much, much harder to unlock) in case the server is stolen and put into some other network.

**Describe the solution you'd like**
Please consider adding a posture check for peer's Public IP-Address(es)/ranges.

**Describe alternatives you've considered**
Alternative is to make a single dedicated, separate vpn channel with IP pinning (site-to-site like in case of IPSEC).

**Comment by piotrekzurek** [comment]
I have added the issue after being refused by the UI to create a posture check based on IP address whereas it actually works already and just needs to have a /32 mask added to the IP address I wanted to PIN the peer to. Kind of obvious but might be worth keeping this thread here in case anyone else looks for it.

Great, great app! Thank you. Looking forward to new features.

**Comment by piotrekzurek** [comment]
Actually when trying to use it with public IP addresses I couldn't. It's possible to create a policy that uses ranges of internal interfaces of the peer, not the public IP addresses (when it's behind NAT), so this would be still a great feature. Reopening...

**Comment by mlsmaycon** [comment]
@piotrekzurek, thanks for your request. We will discuss the feature internally and provide an update once we have a better idea of when we could work on it.

**Comment by spanishdexter** [comment]
Has there been any new developments in regards to this feature? The org I work for, would love to use Netbird to replace our current VPN solution, SoftEther, but this feature is a crucial feature to ensure our fleet of laptops don’t try to connect to the VPN when their already in the office while on the local subnets at our sites.

**Comment by nazarewk** [comment]
> Has there been any new developments in regards to this feature? The org I work for, would love to use Netbird to replace our current VPN solution, SoftEther, but this feature is a crucial feature to ensure our fleet of laptops don’t try to connect to the VPN when their already in the office while on the local subnets at our sites.

You should already be able to achieve your use case with a Posture Check configured with `Peer Network Range` set to your local site's subnet. This will match the subnet EXACTLY (no sub/super-net or overlaps calculations are taken into consideration right now).

**Comment by spanishdexter** [comment]
Yes, I am aware of that, however a lot of our home users on their home networks, use the 192.168.x.x range. This is the same range our offices use. Thats where the VPN can unintentionally get disconnected as well.

**Comment by nazarewk** [comment]
> Yes, I am aware of that, however a lot of our home users on their home networks, use the 192.168.x.x range. This is the same range our offices use. Thats where the VPN can unintentionally get disconnected as well.

This should not be a problem unless you're using **exactly** `192.168.0.0/24` (the default on most home devices), if you're using anything else: `192.168.10.0/24` or even the whole `192.168.0.0/16` you would be fine.

**Comment by spanishdexter** [comment]
Hello so in our testing we have tried this out, it seems to be mostly working well for our other offices (192.168.10.0, 192.168.21.0, etc).

However, our main office is 192.168.0.0/23, and im trying to use a router for testing that is 192.168.0.0/24, to simulate a good sample of our home users personal routers.

Netbird is treating the /23 and /24 the same. I want people at the office to be disconnected while on the 192.168.0.0/23 network and connected when their at home on their 192.168.0.0/24 network. Is there a way to get netbird to tell these aparr, or will I have to rearchitect our office network?

**Comment by spanishdexter** [comment]
Update: I developed a quick work around until thse features are natively included in the client.

Netbird has been awesome and this workaround lets me proceed with using it.

Thanks for building an awesome VPN solution.

https://github.com/spanishdexter/NetbirdIPCheck

**Comment by fourass** [comment]
> Update: I developed a quick work around until these features are natively included in the client.
> 
> Netbird has been awesome and this workaround lets me proceed with using it.
> 
> Thanks for building an awesome VPN solution.
> 
> https://github.com/spanishdexter/NetbirdIPCheck

Hello, hope you're doing well
Can you please share the workaround ? the GitHub link is returning 404

**Comment by spanishdexter** [comment]
@fourass Unfortunately I had to pull the repo, the scripts were not that effective in a real-world production environment, and at this point, I have decided to use Netbird for specific use cases and not for mobile users on laptops at the office where I work, I don't think the script will work that well after my experience, so I decided to pull it. I'll see if I can remove my comment referencing the script. My apologies.

**Comment by PizzaLovingNerd** [comment]
Closing as implemented in https://github.com/netbirdio/netbird/pull/6038. The Peer Network Range posture check now evaluates the public connection IP observed by management in addition to local interface ranges. Use a `/32` prefix for a single IPv4 address or `/128` for a single IPv6 address. See https://docs.netbird.io/manage/access-control/posture-checks/connecting-from-the-office#matching-by-office-public-ip. Cheers!
